### ISSUE TITLE
[REQUEST] Add a function to generate Authorization URL

### ISSUE THREAD

**Issue by lucemans** [MatthiasKunnen/node-itsme-client#5]
Proposed Usage:
```javascript
client.generateAuthURL({redirect_uri: "", scopes: []})
```

Implementation would use the default scopes of "openid" and the project service (ex: "service:name_tst_login")

Proposed Implementation
The base URL for the authorization endpoint is exposed and should be accessible through
`client.idp.configuration.authorization_endpoint`

With the "query-string" library one can easily create the url by doing something like
```
import * as qs from 'query-string';

function generateAuthURL(authconfig) {
   const params = {
      client_id: *REDACTED*,
      response_type: "code",
      scope: "openid " + authconfig.scopes.join(' '),
      redirect_uri: authconfig.redirect_uri,
      state: authconfig.state
   };
   return this.idp.configuration.authorization_endpoint + "?" + qs.stringify(params); 
}
```

Open to any and all feedback

**Comment by lucemans** [comment]
Just noticed that the package `qs` is already used, recommend using that instead.

**Comment by MatthiasKunnen** [comment]
I've prereleased v4.0.0-rc.1 (install with `itsme-client@next`), can you try it out? The examples in the README are updated.
Another question, are you using TypeScript?

**Comment by lucemans** [comment]
Awesome, will do, and yes, I am using typescript. I noticed you had your own linter thing. Personally I havent gotten into linting but, could you maybe explain how yours works and why one should lint?

**Comment by lucemans** [comment]
Just tested it, works flawlessly although I do not comprehend what you did in your example,
In the section `Obtaining user info with an Authorization token`
You use
```javascript
itsmeClient.generateAuthUrl({
            service: 'YOUR_SERVICE_CODE',
            state: 'optional state',
        }),
```
as your `redirect_uri`, I am still just using my normal redirect_uri,
The url generated by `generateAuthUrl` is the url that we redirect the user to, and where they perform the authentication.
Would you want me to write an example on where I am using it right now?

**Comment by MatthiasKunnen** [comment]
Examples are always welcome. Don't you get the correct URL when executing the code in your example?

**Comment by MatthiasKunnen** [comment]
I'm assuming `generateAuthUrl` works as it's supposed to?

**Comment by lucemans** [comment]
Yup! Works perfectly right now

**Comment by MatthiasKunnen** [comment]
Alright, I've released this as 4.0.0 and will be closing this issue. 

Regarding linting, It's used for consistency and to avoid pitfalls. I personally have made my [own presets](https://github.com/MatthiasKunnen/rulebook/tree/master/linters/eslint) which I install in all my projects. I find linting to be a must in all projects, large or small.
